skimage/feature/blob.py

```diff
@@ -339,9 +343,11 @@ def blob_dog(image, min_sigma=1, max_sigma=50, sigma_ratio=1.6, threshold=2.0,
     gaussian_images = [gaussian_filter(image, s) for s in sigma_list]
 
     # computing difference between two successive Gaussian blurred images
-    # multiplying with average standard deviation provides scale invariance
-    dog_images = [(gaussian_images[i] - gaussian_images[i + 1])
-                  * np.mean(sigma_list[i]) for i in range(k)]
+    # to obtain an approximation of the scale invariant Laplacian of the
+    # Gaussian operator
+    dog_images = [
+        (gaussian_images[i] - gaussian_images[i + 1]) for i in range(k)
+    ]
```

Comment:
Thank you so much, @cmarasinou! (Plus I'm not sure what it meant to take the mean of a single element: `np.mean(sigma_list[i])`?)

But where is the division by (sigma_ratio-1), please?
I'm referring to your PR description:

> Simplifying this we can remove the multiplication by the standard deviation and divide by (sigma_ratio-1).

and to the bottom of page 6 in the paper you linked to (https://www.cs.ubc.ca/~lowe/papers/ijcv04.pdf).

Reply:
I think whether or not we divide by `(sigma_ratio - 1)` or any global scale factor will not make any difference.  This is because the following lines just use `peak_local_max` to return the coordinates of the peaks (the absolute magnitude of the peaks are not ever used).

Reply:
Oh, right, right! Thanks, @grlee77 :smile: 
Waiting for CI to complete and merging :shipit:

Reply:
Hi @mkcor, I will need to make a followup to this revisiting the absolute scaling. I had not noticed that there is also an absolute threshold being passed onto `peak_local_max` as well, so it will be important in terms of the appropriate setting of the user-parameter `threshold`. Also, the docstring example now longer matches after this PR, but this was not caught due to #5502!

For reference this is what `image_cube` looks like for the docstring example prior to this PR:
![cube_old](https://user-images.githubusercontent.com/6528957/128531335-140c0a62-b966-4dd4-adc0-b085a90689e0.png)

vs. after this PR where you can see that there is less bias toward larger scales
![cube_new](https://user-images.githubusercontent.com/6528957/128531444-138daca4-9ca2-430c-b4ac-450e4cc9ed5a.png)

To avoid small scale edges being detected we will probably have to increase `min_sigma` substantially for the docstring example and fix the threshold.

Reply:
Oops! I'm diving deeper and it's a little scary, indeed... Reviewing your PR #5503 now!